src/raftstore/store/store.rs

```diff
@@ -1290,6 +1290,63 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         // we will call the callback with timeout error.
     }
 
+    fn propose_batch_raft_snapshot_command(&mut self,
+                                           batch: Vec<RaftCmdRequest>,
+                                           on_finished: BatchCallback) {
+        let size = batch.len();
+        let mut ret: Vec<Option<RaftCmdResponse>> = vec![None; size];
+        let mut batch_sizes: HashMap<_, u64> = HashMap::default();
+        let mut cache: HashMap<u64, Option<Option<RaftCmdResponse>>> = HashMap::default();
+        for (idx, msg) in batch.into_iter().enumerate() {
+            let mut err_resp = RaftCmdResponse::new();
+
+            if let Err(e) = self.validate_store_id(&msg) {
+                bind_error(&mut err_resp, e);
+                ret[idx] = Some(err_resp);
+                continue;
+            }
+
+            if let Err(e) = self.validate_region(&msg) {
+                bind_error(&mut err_resp, e);
+                ret[idx] = Some(err_resp);
+                continue;
+            }
+
+            let region_id = msg.get_header().get_region_id();
+            let mut peer = self.region_peers.get_mut(&region_id).unwrap();
+
+            let mut count = batch_sizes.entry(region_id).or_insert(0);
+            *count += 1;
+            let mut resp = cache.entry(region_id).or_insert_with(|| None);
```

Comment:
I don't think we need this cache here, calling propose_snapshot for the request is fast.